CONTRIBUTING.md

```diff
@@ -44,24 +46,68 @@ go test -timeout=30s -p 1 ./...
+PERCONA_TEST_CHECKS_INTERVAL=10s # set it to any low duration to execute checks quickly
+```
+* Setup the devcontainer using `make env-up` and run your changes inside it by running `make env` and then`make run`.
+* Follow the steps in the previous section to set up instances for monitoring.
+* Go to the PMM dashboard and enable STT in `PMM Settings -> Advanced Settings`.
+* Any failed checks will produce result on the dashboard.
+
+## Working with Integrated Alerting
+* Make sure you have instances setup up for monitoring.
+* First we need to setup global communication settings for notification channels. Go to "Communications" tab under "PMM Settings" and set either the SMTP config for email alerts or the Slack webhook URL for slack alerts or both.
+* Go to the "Integrated Alerting" dashboard and select the "Notification Channels" tab. Create a notification channel to send the alerts to.
+* Now create an alert rule using the built-in alert rule templates and the notification channel.
+* If the alert rule condition satisfies then the alert will be visible on the "Alerts" tab and notification sent to the channel provided that it was configured correctly.
 
 # Internals
 
 There are three makefiles: `Makefile` (host), `Makefile.devcontainer`, and `Makefile.include` (included in both). `Makefile.devcontainer` is mounted on top of `Makefile` inside devcontainer (see `docker-compose.yml`) to enable `make env TARGET=target-name` usage.
 
 Devcontainer initialization code is located in `.devcontainer/setup.py`. It uses multiprocessing to run several commands in parallel to speed-up setup.
 
+## Code Structure
+```
+.
+├── bin - binaries
+├── cmd - code for any scripts run by managed
+├── data - integrated alerting templates and generated code
+├── models - database helpers and types, the database schema can be found in models/database.go file
+├── services - contains all the APIs for interacting with services like alertmanager, checks service, victoriametrics, etc
+├── testdata - dummy data files used in unit tests
+├── utils - utilities
+```
 
 # How to make PR
+* If the changes require multiple PRs spanning multiple repos make sure to keep the branch names same.
+* If the PR requires any API changes then make your changes in `PMM-2.0` branch of the [API repo](https://github.com/percona/pmm) and pull those changes in your pmm-managed branch by mentioning the API changes branch name in the `Gopkg.toml` constraint and running `dep ensure -v -update github.com/percona/pmm`.
+* If the PR introduces changes to the IA templates make sure to run `make gen` to udpate the generated code.
+* If the PR changes any files named `deps.go` make sure to run `make gen` to generate mock clients.
+* In case a new package is added make sure to udpate `packages.dot` file.
 
 Before making PR, please run these commands locally:
 * `make env TARGET=check-all` to run all checkers and linters.
 * `make env TARGET=test-race` to run tests.
```

Comment:
```suggestion
* `make env TARGET=test-race` to run tests.
* For help, please post on the [PMM 2.x Forums](https://forums.percona.com/c/percona-monitoring-and-management-pmm/percona-monitoring-and-management-pmm-v2/)
```

Reply:
@shoffman-percona i think our community slack would be a better way for the contributors to reach out to us, should I add that instead or along with the forum link?

Reply:
My fear of using slack is it sets the impression we are monitoring it...and I don't know who is (I go there on occasion but for now the forums are FAR more active than the community slack

Reply:
done